packages/suite/src/views/wallet/send/Outputs/TokenSelect/TokenSelect.tsx

```diff
@@ -0,0 +1,412 @@
+} from '@trezor/components';
+import { selectCurrentFiatRates, updateFiatRatesThunk } from '@suite-common/wallet-core';
+import { Account, Timestamp, TokenAddress } from '@suite-common/wallet-types';
+import {
+    CoinLogo,
+    SearchAsset,
+    SelectAssetModal,
+    TokenTabs,
+    TokenTab,
+    AssetProps,
+    ITEM_HEIGHT,
+} from '@trezor/product-components';
+import {
+    getContractAddressForNetwork,
+    getTokenExplorerUrl,
+    hasNetworkFeatures,
+} from '@suite-common/wallet-utils';
+import { Elevation, mapElevationToBackground, spacings } from '@trezor/theme';
+import {
+    selectCoinDefinitions,
+    selectIsSpecificCoinDefinitionKnown,
+    TokenDefinitions,
+} from '@suite-common/token-definitions';
+import { getCoingeckoId, networks, NetworkSymbol } from '@suite-common/wallet-config';
+import { FiatCurrencyCode } from '@suite-common/suite-config';
+import { TokenInfo } from '@trezor/blockchain-link-types';
+
+import { selectIsCopyAddressModalShown } from 'src/reducers/suite/suiteReducer';
+import {
+    FiatValue,
+    FormattedCryptoAmount,
+    HiddenPlaceholder,
+    Translation,
+} from 'src/components/suite';
+import { SUITE } from 'src/actions/suite/constants';
+import { selectLocalCurrency } from 'src/reducers/wallet/settingsReducer';
+import {
+    enhanceTokensWithRates,
+    getTokens,
+    sortTokensWithRates,
+} from 'src/utils/wallet/tokenUtils';
+import { useDispatch, useSelector, useTranslation } from 'src/hooks/suite';
+import { useSendFormContext } from 'src/hooks/wallet';
+import { TokenAddressRow } from 'src/components/suite/copy/TokenAddressRow';
+import { copyAddressToClipboard, showCopyAddressModal } from 'src/actions/suite/copyAddressActions';
+
+// eslint-disable-next-line local-rules/no-override-ds-component
+export const TokenSelectContainer = styled(Card)<{ $isDisabled: boolean; $elevation: Elevation }>`
+    background: ${({ theme, $elevation }) => mapElevationToBackground({ $elevation, theme })};
+    ${({ $elevation, theme }) =>
+        $elevation === 1 &&
+        css`
+            box-shadow: ${theme.boxShadowBase};
+        `}
+    cursor: ${({ $isDisabled }) => ($isDisabled ? 'default' : 'pointer')};
+`;
+
+export const CursorWrapper = styled.div<{ $isDisabled: boolean }>`
+    cursor: ${({ $isDisabled }) => ($isDisabled ? 'not-allowed' : 'pointer')};
+`;
+
+const getToken = (token: TokenInfo, symbol: Account['symbol'], shouldTryToFetch: boolean) => ({
+    symbol: token.symbol ?? '',
+    networkSymbol: symbol,
+    cryptoName: token.name,
+    badge: shouldTryToFetch ? undefined : <Translation id="TR_UNRECOGNIZED" />,
+    coingeckoId: getCoingeckoId(symbol) ?? '',
+    contractAddress: token.contract,
+    shouldTryToFetch,
+    height: ITEM_HEIGHT,
+});
+
+const buildTokenOptions = (
+    accountTokens: Account['tokens'],
+    symbol: Account['symbol'],
+    coinDefinitions: TokenDefinitions['coin'],
+    activeTokenTab: TokenTab['tab'],
+): AssetProps[] => {
+    const result = [];
```

Comment:
this is implicitly typed as `any`
```suggestion
    const result: AssetProps[] = [];
```